app/assets/javascripts/components/overview/available_actions.jsx

```diff
@@ -54,15 +54,19 @@ const AvailableActions = ({ course, current_user, updateCourse, courseCreationNo
   };
 
   const deleteCourseFunc = () => {
-    // The action is only available once a course has been removed from all campaigns.
-    if (course.published) {
-      return alert(I18n.t('courses.delete_course_instructions'));
-    }
-
+    const courseSlug = course.slug;
     const enteredTitle = prompt(I18n.t('courses.confirm_course_deletion', { title: course.title }));
     // Check if enteredTitle is not null before calling trim.
     if (enteredTitle !== null && enteredTitle.trim() === course.title.trim()) {
-      return dispatch(deleteCourse(course.slug));
+      // If in Wiki Ed or course has no campaigns, delete the course directly; otherwise, remove from campaign first.
+      if (Features.wikiEd || !course.campaigns || course.campaigns.length === 0) {
```

Comment:
Is the check for Features.wikiEd here redundant, since it will only render the button in the first place if the campaigns length is zero (because that means it is not published)?

Reply:
That is true! Another oversight, since published campaigns are automatically added to Miscellanea. I'll remove that now